components/scream/src/physics/shoc/tests/shoc_run_and_cmp.cpp

```diff
@@ -0,0 +1,205 @@
+#include "share/scream_session.hpp"
+#include "share/util/file_utils.hpp"
+#include "share/util/scream_utils.hpp"
+#include "share/scream_types.hpp"
+#include "share/scream_assert.hpp"
+
+#include "physics/shoc/shoc_f90.hpp"
+#include "physics/shoc/shoc_functions_f90.hpp"
+#include "physics/shoc/shoc_ic_cases.hpp"
+
+#include <vector>
+
+namespace {
+using namespace scream;
+using namespace scream::util;
+using namespace scream::shoc;
+
+template <typename Scalar>
+static Int compare (const std::string& label, const Scalar* a,
+                    const Scalar* b, const Int& n, const Real& tol) {
+  Int nerr = 0;
+  Real den = 0;
+  for (Int i = 0; i < n; ++i)
+    den = std::max(den, std::abs(a[i]));
+  Real worst = 0;
+  for (Int i = 0; i < n; ++i) {
+    if (std::isnan(a[i]) || std::isinf(a[i]) ||
+        std::isnan(b[i]) || std::isinf(b[i])) {
+      ++nerr;
+      continue;
+    }
+    const auto num = std::abs(a[i] - b[i]);
+    if (num > tol*den) {
+      ++nerr;
+      worst = std::max(worst, num);
+    }
+  }
+  if (nerr) {
+    std::cout << label << " nerr " << nerr << " worst " << (worst/den)
+              << " with denominator " << den << "\n";
+  }
+  return nerr;
+}
+
+Int compare (const std::string& label, const double& tol,
+             const FortranData::Ptr& ref, const FortranData::Ptr& d) {
+  Int nerr = 0;
+  FortranDataIterator refi(ref), di(d);
+  scream_assert(refi.nfield() == di.nfield());
+  for (Int i = 0, n = refi.nfield(); i < n; ++i) {
+    const auto& fr = refi.getfield(i);
+    const auto& fd = di.getfield(i);
+    scream_assert(fr.size == fd.size);
+    nerr += compare(label + std::string(" ") + fr.name,
+                    fr.data, fd.data, fr.size, tol);
+  }
+  return nerr;
+}
+
+struct Baseline {
+  Baseline () {
+    for (const int nadv : {1, 6})
+      params_.push_back({ic::Factory::standard, 1, 72, 1, nadv, 300});
```

Comment:
1. I'd like this to be dtime=150, nadv=15.
2. We want to test on >1 column.
3. Also run with >1 tracer.
See point 4 in another comment.